controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -95,6 +95,12 @@ func (c *Cluster) controlPlaneIsHealthy(ctx context.Context) (healthCheckResult,
 			Namespace: metav1.NamespaceSystem,
 			Name:      staticPodName("kube-apiserver", name),
 		}
+
+		if err = checkNodeNoExecuteCondition(&node); err != nil {
```

Comment:
```suggestion
		if err := checkNodeNoExecuteCondition(&node); err != nil {
```

Could we just return true/false here instead of an error?

Reply:
Pod health checks return error and returned error is used by the caller functions. So, just to be consistent with the rest, I'd keep it like this.

Reply:
@vincepri that ok?